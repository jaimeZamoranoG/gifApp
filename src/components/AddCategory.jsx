import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('');

    const onInputChange = ({target})=>{
        setinputValue(target.value);
    }

    const onSubmit = (event)=>{
        event.preventDefault();
        const newValue=inputValue.trim();
        if (newValue.length <=1) return;
        onNewCategory(newValue);
        setinputValue("");
    }

    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input
            type='text'
            placeholder='buscar gifs'
            value={inputValue}
            onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}