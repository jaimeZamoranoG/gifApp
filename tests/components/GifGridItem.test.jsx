import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../src/components/GifGridItem"

describe("Pruebas en GifGridItem",()=>{

    const title = 'Cookie Monster Hello GIF by Sesame Street';
    const url = 'https://media2.giphy.com/media/GRPy8MKag9U1U88hzY/giphy.gif?cid=381d12a0louw4bwtfpmxgk57uw5jofj4sz5mkp4gke567sol&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test("Debe estar como el snapshot",()=>{
        const {container}  = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con url y alt indicado',()=>{
        render(<GifGridItem title={title} url={url} />);
        //screen.debug();
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('debe estar el titulo como texto',()=>{
        render(<GifGridItem title={title} url={url} />);
        expect(screen.findByText(title)).toBeTruthy();
    });

})