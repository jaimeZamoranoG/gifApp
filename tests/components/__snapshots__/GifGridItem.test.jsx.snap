// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe estar como el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Cookie Monster Hello GIF by Sesame Street"
      src="https://media2.giphy.com/media/GRPy8MKag9U1U88hzY/giphy.gif?cid=381d12a0louw4bwtfpmxgk57uw5jofj4sz5mkp4gke567sol&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Cookie Monster Hello GIF by Sesame Street
    </p>
  </div>
</div>
`;
